Library/Homebrew/utils/pypi.rb

```diff
@@ -249,9 +249,16 @@ def self.update_python_resources!(formula, version: nil, package_name: nil, extr
       end
     end
 
+    python_deps = formula.deps
+                         .map(&:to_formula)
+                         .select { |f| f.name.start_with?("python@") }
```

Comment:
```suggestion
                         .select { |f| f.name.start_with?("python@") }
                         .map(&:to_formula)
```

as `to_formula` is somewhat expensive and can fail

Reply:
done, had to switch to a regex to handle `uses_from_macos "python"`